### PR TITLE
Metrics in MQTT and REST Adapters

### LINKED ISSUE #280: Allow metrics in all our services, implement in adapers
* Move the reporters in the service-base module
* Adapters: Define and implement base metrics, that are useful for all adapters
  * number of messages (telemetry/event), that the adapters receive
  * number of messages (telemetry/event), that the adapters forwards to hono messaging 
  * number of actual connections to devices

### PR DESCRIPTION
**Pull request by pellmann**
Metrics for processed and undelivered messages in MQTT and REST Adapter
- Metrics are written to InfluxDB
- Metrics are shown in Grafana default board
- Templates for InfluxDB

Fixes #280 and #309

Signed-off-by: pellmann <marc.pellmann@bosch-si.com>

### CODE REVIEW

File: adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java [hunk 1]
```diff
@@ -48,9 +49,20 @@
     private static final int IANA_MQTT_PORT = 1883;
     private static final int IANA_SECURE_MQTT_PORT = 8883;
 
-    private MqttServer server;
-    private MqttServer insecureServer;
+    private MqttServer                server;
+    private MqttServer                insecureServer;
     private Map<MqttEndpoint, String> registrationAssertions = new HashMap<>();
+    private MqttAdapterMetrics        metrics;
```
Comment (ppatierno) [inline adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java]: @pellmann why are we moving to this approach, aligning member names ?

Comment (pellmann) [inline adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java]: I think it was the definition of our code style. Is it not? My IDE does this, when I format the code according to our styles.

Comment (pellmann) [inline adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java]: Oh - perfect.. I personally also do not like it. :) So I will change my settings..

Comment (ppatierno) [inline adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java]: Thanks ! I have just reverted to previous members declaration alignment.

**Pull request merged**